code/modules/mob/living/simple_animal/hostile/megafauna/captainarlem.dm

```diff
@@ -152,29 +214,42 @@ mob/living/simple_animal/hostile/megafauna/captainarlem/do_attack_animation(atom
 	P.fire(set_angle)
 
 /mob/living/simple_animal/hostile/megafauna/captainarlem/proc/blast(set_angle)
+	if(blast_cooldown >= world.time)
+		return
+	say("Burn in hell!", spans = spans_list, sanitize = FALSE)
+	blast_cooldown = world.time + blast_cooldown_time
 	var/turf/target_turf = get_turf(target)
 	playsound(src, 'sound/weapons/lasercannonfire.ogg', 200, 1, 2)
 	newtonian_move(get_dir(target_turf, src))
 	var/angle_to_target = Get_Angle(src, target_turf)
 	if(isnum(set_angle))
 		angle_to_target = set_angle
-	var/static/list/plasmacaster_angles = list(12.5, 7.5, 2.5, -2.5, -7.5, -12.5)
-	for(var/i in plasmacaster_angles)
-		fire_plasmacaster(target_turf, angle_to_target + i)
+	var/static/list/colossus_shotgun_shot_angles = list(15, 7.5, 0, -7.5, -15)
+	for(var/i in colossus_shotgun_shot_angles)
+		shoot_projectile(target_turf, angle_to_target + i)
+
 
 /mob/living/simple_animal/hostile/megafauna/captainarlem/proc/eyebots()
+	if(eyebots_cooldown >= world.time)
+		return
+	say("Need some aid!", spans = spans_list, sanitize = FALSE)
+	eyebots_cooldown = world.time + eyebots_cooldown_time
 	visible_message("<span class='danger'>[src] presses a button on their wrist, activating some of the eyebots!</span>")
-	for(var/obj/effect/decal/remains/deadeyebot/H in range(src, 10))
-		if(prob(40))
+	for(var/mob/living/simple_animal/hostile/eyebot/botty in range(src, 7))
+		if(prob(50))
+			new /mob/living/simple_animal/hostile/eyebot(botty.loc)
+			qdel(botty)
+	for(var/obj/effect/decal/remains/deadeyebot/H in range(src, 8))
+		if(prob(50))
 			new /mob/living/simple_animal/hostile/eyebot(H.loc)
-	
-/mob/living/simple_animal/hostile/megafauna/captainarlem/proc/self_destruct()
-	explosion(src,3,5,7,7)
-	qdel(src)
+			qdel(H)
 
 /mob/living/simple_animal/hostile/megafauna/captainarlem/death()
 	do_sparks(3, TRUE, src)
+	playsound(src, 'sound/weapons/armbomb.ogg', 75, 1, -3)
 	addtimer(CALLBACK(src, .proc/self_destruct), 4 SECONDS)
 	return ..()
 
-
+/mob/living/simple_animal/hostile/megafauna/captainarlem/proc/self_destruct()
+	explosion(src,3,5,7,7)
+	qdel(src)
```

Comment:
Put checks here, such as if the mob wasn't qdeleted in the meantime already.
Given it's a timed action, that could have happened.